internal/fsutil.go

```diff
@@ -72,6 +73,44 @@ func createTestExtImg(ctx context.Context, t *testing.T, extName string, testFil
 	return imgFile.Name()
 }
 
+// Devicefile represents a block device
+type Devicefile struct {
+	Subpath string
+	UID     int
+	GID     int
+	Dev     int // major number
+}
+
+// CreateBlockDevice creates a block device, or block special file for testing
+func CreateBlockDevice(ctx context.Context, t *testing.T, extName string, testFile Devicefile) string {
+	t.Helper()
+	tempdir, err := ioutil.TempDir("", "")
+	require.NoError(t, err, "failed to create temp dir for ext img")
+
+	deviceFile := filepath.Join(tempdir, testFile.Subpath)
+	if _, err := os.Stat(deviceFile); err == nil {
+		os.Remove(deviceFile)
+		require.NoError(t, err, "failed to remove duplicate block device")
+	}
+
+	os.MkdirAll(filepath.Dir(deviceFile), 0750)
+	require.NoError(t, err, "failed to mkdir for the block device")
+
+	err = syscall.Mknod(deviceFile, syscall.S_IFBLK, testFile.Dev)
+	require.NoError(t, err, "failed to create a new block device")
+
+	os.Chmod(deviceFile, 0600)
+	require.NoError(t, err, "failed to change file mode for the new created block device")
+
+	os.Chown(deviceFile, testFile.UID, testFile.GID)
+	require.NoError(t, err, "failed to grant permission to the new created block device")
+
+	_, err = exec.CommandContext(ctx, "mkfs."+extName, deviceFile).CombinedOutput()
+	require.NoError(t, err, "Can not mkfs.ext4 for the new block device")
```

Comment:
You cannot call out mkfs.ext4 since extName can be something else. That being said, I think you can remove extName parameter. We only use ext4 right now.

Reply:
Fixed.